### PR TITLE
Gifts

### PR DESCRIPTION
**Pull request by PavelBulat**

### CODE REVIEW

File: modules/features/falcon_feature_gifts/falcon_feature_gifts.info.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: 'Falcon Feature Gifts'
+description: 'Provides the Gifts'
```
Comment (kalabro) [inline modules/features/falcon_feature_gifts/falcon_feature_gifts.info.yml]: We need to update it to be more descriptive. I'd say "Charity gifts shop" or something like this.

File: modules/features/falcon_feature_gifts/config/install/commerce_product.commerce_product_type.gift.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+langcode: en
+status: true
+dependencies: {  }
+id: gift
+label: Gift
+description: ''
```
Comment (kalabro) [inline modules/features/falcon_feature_gifts/config/install/commerce_product.commerce_product_type.gift.yml]: Missing description.

**Review by kalabro — APPROVED** [review]

File: tests/api/commerce_decoupled_checkout/GiftCest.php [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    $I->seeResponseCodeIs(HttpCode::CREATED);
+  }
+
+  /**
+   * Successful single gift via Direct Debit Test payment gateway - SEPA.
+   *
+   * @param \ApiTester $I
+   * @group additional
+   */
+  public function giftRecurringDirectDebitSEPASuccess(\ApiTester $I) {
```
Comment (kalabro) [inline tests/api/commerce_decoupled_checkout/GiftCest.php]: Can you please drop all recurring test cases? Recurring payments are not applicable to Gifts. Gifts order is one-off purchase.

**Review by kalabro — DISMISSED** [review]

File: tests/api/commerce_decoupled_checkout/GiftCest.php [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+   * @group additional
+   */
+  public function giftSingleExampleSuccess(\ApiTester $I) {
+    $I->amGoingTo('Post correct order Gift to Commerce Create REST endpoint.');
+    $I->haveHttpHeader('Content-Type', 'application/json');
+
+    $post = $this->post;
+    $I->sendPOST('/commerce/order/create', $post);
+
+    /** @var \Drupal\commerce_store\StoreStorageInterface $commerce_store */
```
Comment (kalabro) [inline tests/api/commerce_decoupled_checkout/GiftCest.php]: Can you please remove commented-out code?

File: tests/api/commerce_decoupled_checkout/GiftCest.php [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+    $stores = $entity_type_manager->getStorage('commerce_store')->loadByProperties(['type' => 'online']);
+
+    // Create product variation gift.
+    $this->variation = $entity_type_manager->getStorage('commerce_product_variation')->create([
+      'type' => 'gift',
+      'title' => 'Test gift',
+      'sku' => 'test gift',
+      'status' => 1,
+      'price' => new Price('0', 'USD'),
```
Comment (kalabro) [inline tests/api/commerce_decoupled_checkout/GiftCest.php]: Why does it have zero price?

File: tests/api/commerce_decoupled_checkout/GiftCest.php [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+   * @var array
+   */
+  private $post = [
+    'order' => [
+      'type' => 'gift',
+      'order_items' => [
+        [
+          'type' => 'gift',
+          'unit_price' => [
+            'number' => 15,
```
Comment (kalabro) [inline tests/api/commerce_decoupled_checkout/GiftCest.php]: Why does it have this price?

File: tests/api/commerce_decoupled_checkout/GiftCest.php (hunk 3 above)
Comment (PavelBulat) [reply tests/api/commerce_decoupled_checkout/GiftCest.php]: Ok, I did not know about recurring payments. =)

File: tests/api/commerce_decoupled_checkout/GiftCest.php (hunk 5 above)
Comment (PavelBulat) [reply tests/api/commerce_decoupled_checkout/GiftCest.php]: i changed to 10, but i think it doesn`t metter for this case.

File: tests/api/commerce_decoupled_checkout/GiftCest.php (hunk 6 above)
Comment (PavelBulat) [reply tests/api/commerce_decoupled_checkout/GiftCest.php]: Today i looked to commerce_decoupled_checkout module and i think, the unit_price don`t need in this case.

File: tests/api/commerce_decoupled_checkout/GiftCest.php (hunk 4 above)
Comment (PavelBulat) [reply tests/api/commerce_decoupled_checkout/GiftCest.php]: Fixed.

File: modules/features/falcon_feature_gifts/config/install/commerce_product.commerce_product_type.gift.yml (hunk 2 above)
Comment (PavelBulat) [reply modules/features/falcon_feature_gifts/config/install/commerce_product.commerce_product_type.gift.yml]: Fixed.

File: modules/features/falcon_feature_gifts/falcon_feature_gifts.info.yml (hunk 1 above)
Comment (PavelBulat) [reply modules/features/falcon_feature_gifts/falcon_feature_gifts.info.yml]: Fixed.

**Review by spleshka — APPROVED** [review]

**Pull request merged**